commands/transactions.go

```diff
@@ -0,0 +1,25 @@
+package commands
+
+import (
+	"github.com/bitcoin-sv/spv-wallet-go-client/internal/api/v1/user/querybuilders"
+	"github.com/bitcoin-sv/spv-wallet/models/response"
+)
+
+// RecordTransaction holds the arguments required to record a user transaction.
+type RecordTransaction struct {
+	Metadata    querybuilders.Metadata `json:"metadata"`    // Metadata associated with the transaction.
+	Hex         string                 `json:"hex"`         // Hexadecimal string representation of the transaction.
+	ReferenceID string                 `json:"referenceId"` // Reference ID for the transaction.
+}
+
+// DraftTransaction holds the arguments required to create user draft transaction.
+type DraftTransaction struct {
+	Config   response.TransactionConfig `json:"config"`   // Configuration for the transaction.
+	Metadata querybuilders.Metadata     `json:"metadata"` // Metadata related to the transaction.
+}
+
+// UpdateTransactionMetadata holds the arguments required to update a user transaction's metadata.
+type UpdateTransactionMetadata struct {
+	ID       string                 `json:"-"`        // Unique identifier of the transaction to be updated.
```

Comment:
Maybe, if you decided to put comments for each field - please describe why the `ID` is ignored in json.